docs/source/azure/azurefunctions.rst

```diff
@@ -0,0 +1,52 @@
+.. azurefunctions:
+
+Azure Functions Support
+-----------------------
+
+Cloud Custodian Integration
+===========================
+Support of Azure Functions in Cloud Custodian is still in progress.
+
+
+Provision Options
+#################
+
+When running in Azure functions, a storage account, application insights instance, and an app service
+plan is provisioned in your subscription to enable running the functions in an app service.
+
+Execution in Azure functions comes with a default set of configurations for the provisioned
+resources. To override these setting you must set 'provision-options' with one of the following
+keys:
+
+- location (default: West US 2)
+- appInsightsLocation (default: West US 2)
+- sku (default: Standard)
+- skuCode (default: S1)
+- workerSize (default: 0)
+
+The location allows you to choose the region to deploy the resource group and resources that will be
+provisioned. Application Insights has six available locations and thus can not always be in the same
+region as the other resources: West US 2, East US, North Europe, South Central US, Southeast Asia, and
+West Europe. The sku, skuCode, and workerSize correlate to scaling up the App Service Plan.
+
+An example on how to set the provision-options when running in azure-functions mode:
+
+.. code-block:: yaml
+
+    policies:
+      - name: stopped-vm
+        mode:
+            type: azure-functions
+            provision-options:
+              location: East US
```

Comment:
So I feel like this schema needs some way to identify the App Service.  We need some way to support multiple policies in a single app service so my assumption is you'd provide the name and if it already exists we would just deploy the function to it, otherwise create it.   This has plenty of issues so I'm not attached to the strategy if there is a better choice.   

Can not do one app service per policy though as that'd be ~$15/month/policy (and you'd hit the app service limit before you got a useful number of policies out anyway).

Reply:
oh another idea is to use a common app service plan and run one app service per policy in the plan - basic plans support unlimited apps.  It is extra overhead on the server but it might dramatically simplify things until consumption public preview.  In this case you'd need the plan name somehow.  That would be $55/month but unlimited apps/functions/policies.  I think the shared one for $10/month lets you do 100 functions/policies which isn't untenable in the short term.

Reply:
Per offline discussion - will allow user to specify app service plan.